packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2834,3 +2882,277 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+typedef SelectionListenerSelectionChangedCallback = void Function(List<SelectedContentRange<Object>>? selections);
+
+/// A widget that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in its subtree.
+///
+/// {@tool dartpad}
+/// This example shows how to color red the active selection
+/// under a [SelectionArea] or [SelectableRegion].
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.1.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to replace the active selection
+/// under a [SelectionArea] or [SelectableRegion] with a widget.
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.2.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [SelectionArea] which provides an overview of the selection system.
+///   * [SelectableRegion] which provides an overview of the selection system.
+class SelectionListener extends StatefulWidget {
+  /// Create a new [SelectionListener] widget.
+  const SelectionListener({
+    super.key,
+    required this.onSelectionChanged,
+    required this.child,
+  });
+
+  /// Called when the user changes the selection of its child.
```

Comment:
Related to my questions above about this class, I would be more specific about "its child". Is it just the child widget, or anything in that subtree, etc.?

Reply:
Updated the docs to be more clear about this.